compiler/rustc_interface/src/interface.rs

```diff
@@ -58,6 +58,11 @@ impl Compiler {
     }
 }
 
+#[allow(rustc::bad_opt_access)]
```

Comment:
Is `set_parallel_mode` still the best name here? It doesn't match `sync::set_dyn_thread_safe_mode`.

Reply:
renamed it to `set_thread_safe_mode`